### PR TITLE
Fix MaxRequestBodySize in IIS

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/dotnet/aspnetcore/issues/24555
Fixes https://github.com/dotnet/aspnetcore/issues/22950
Fixes https://github.com/dotnet/aspnetcore/issues/22834

- When checking the MaxRequestBodySize, we never updated it with the value from IIS.
- Change int to uint to allow uint.MaxValue
- Remove 30MB constant in managed code as the value already defaults to 30MB from native and we always set the value in managed from native.

# Description
There is two bug in the IIS MaxRequestBodySize feature which made it so it wouldn't read the value from IIS correctly and couldn't accept a value larger than Int.MaxValue. This made the middleware too restrictive, causing people difficulties who were trying to increase the limit.

# Customer Impact

Fixes the feature by reading the correct value from native as well as accepting UInt.MaxValue, matching the max limit in IIS.

# Regression?

No

# Risk

Low. This fixes an already existing feature and people who were working around it today wouldn't be affected.

### CODE REVIEW

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs [hunk 1]
```diff
@@ -54,6 +54,72 @@ public async Task SetIISLimitMaxRequestBodySizeE2EWorks()
...
+                        .GetOrAdd("security")
+                        .GetOrAdd("requestFiltering")
+                        .Remove();
+                });
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var result = await deploymentResult.HttpClient.PostAsync("/ReadRequestBodyLarger", new StringContent(new string('a', 10)));
+
+            // IIS either returns a 404 or a 413 based on versions of IIS.
+            // Check for both as we don't know which specific patch version.
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs:119]:
```suggestion
```

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs [hunk 2]
```diff
@@ -54,6 +54,72 @@ public async Task SetIISLimitMaxRequestBodySizeE2EWorks()
...
+                        .GetOrAdd("security")
+                        .GetOrAdd("requestFiltering")
+                        .GetOrAdd("requestLimits", "maxAllowedContentLength", "4294967295");
+                });
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var result = await deploymentResult.HttpClient.PostAsync("/ReadRequestBodyLarger", new StringContent(new string('a', 10000)));
+
+            // IIS either returns a 404 or a 413 based on versions of IIS.
+            // Check for both as we don't know which specific patch version.
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs:97]:
```suggestion
```

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs [hunk 3]
```diff
@@ -54,6 +54,72 @@ public async Task SetIISLimitMaxRequestBodySizeE2EWorks()
...
+                        .GetOrAdd("security")
+                        .GetOrAdd("requestFiltering")
+                        .GetOrAdd("requestLimits", "maxAllowedContentLength", "100000000");
+                });
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var result = await deploymentResult.HttpClient.PostAsync("/ReadRequestBodyLarger", new StringContent(new string('a', 100000000)));
+
+            // IIS either returns a 404 or a 413 based on versions of IIS.
+            // Check for both as we don't know which specific patch version.
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs:75]:
```suggestion
```

**Review by Tratcher — APPROVED** [review]

**Review by BrennanConroy — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs [hunk 4]
```diff
@@ -54,6 +54,66 @@ public async Task SetIISLimitMaxRequestBodySizeE2EWorks()
...
+        public async Task SetIISLimitMaxRequestBodySizeE2EWorksNoLimit()
+        {
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters();
+            deploymentParameters.ServerConfigActionList.Add(
+                (config, _) => {
+                    config
+                        .RequiredElement("system.webServer")
+                        .GetOrAdd("security")
+                        .GetOrAdd("requestFiltering")
+                        .Remove();
```
Comment (halter73) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs:108]: What is this actually removing? I don't see any reference to requestLimits.

**Review by halter73 — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs (hunk 4 above)
Comment (jkotalik) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/MaxRequestBodySizeTests.cs:108]: Yeah this test is redundant; I'll delete it.

**Comment by pruchita** [comment]
I want to upload a file of size greater than UInt.MaxValue. Is there any way I can do that? I get a 413.1- Request entity too large error when uploading file larger than 4gb in asp.net core.

**Comment by jkotalik** [comment]
You'd need to break up the request body into pieces, See [Stack Overflow](https://stackoverflow.com/questions/16958448/what-is-http-multipart-request#:~:text=A%20HTTP%20multipart%20request%20is%20a%20HTTP%20request,HTTP%20clients%20to%20upload%20files%20to%20the%20server) and [ASP.NET Core Docs](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/file-uploads?view=aspnetcore-5.0#upload-large-files-with-streaming)

**Pull request merged**